tests/sla_tests.js

```diff
@@ -619,5 +709,387 @@ describe('SLA Bridge and Channels Tester', function() {
+    var hangupRequested = false;
+
+    nonStationHangsup();
+    setTimeout(function () {
+      secondChannel = getMockChannel();
+      secondChannel.name = 'SIP/phone2';
+      secondChannel.caller = {'number': '5678'};
+      secondChannel.outbound = true;
+      var sla2 = require('../lib/sla.js')(client, config, secondChannel, '42')
+      .catch(errHandler)
+      .done();
+    }, asyncDelay * 7);
+
+    function nonStationHangsup() {
+      setTimeout(function() {
+        if (channels.length === 0 && bridgeChannels.length === 0 &&
+          channel.outbound && channel.wasAnswered && channel.wasHungup &&
+          secondChannel.outbound && secondChannel.wasAnswered && 
+          secondChannel.wasHungup && dialed[0].wasAnswered &&
+          dialed[0].nonStation && dialed[0].wasHungup) {
+          done();
+        } else {
+
+          if(toDial[index]) {
+            channel.emit('ChannelDtmfReceived', {digit: toDial[index],
+              channel: channel}, {channel: channel});
+            index += 1;
+          }
+
+          if (secondChannel && channel && bridgeChannels.length === 3 &&
+            !hangupRequested) {
+              hangupRequested = true;
+            dialed[0].hangup(function() {});
+          }
+          nonStationHangsup();
+        }
+      }, asyncDelay);
+    }
+  });
+
+  it('test whether or not a station channel can join the extension after it ' +
+     'has been on hold',
+      function(done) {
+    var client = getMockClient();
+    var channel = getMockChannel();
+    var stationReturns;
+    channel.name = 'SIP/phone3';
+    channel.caller = {'number': '1234'};
+    channel.inbound = true;
+
+    var sla = require('../lib/sla.js')(client, config, channel, '42')
+      .catch(errHandler)
+      .done();
+
+    extensionUnhold();
+
+    function extensionUnhold() {
+      setTimeout(function() {
+        if (channels.length === 2 && bridgeChannels.length === 2 &&
+          channel.inbound && channel.wasAnswered && !channel.wasHungup &&
+          dialed[0].wasAnswered && !dialed[0].nonStation &&
+          dialed[0].wasHungup && stationReturns.wasAnswered &&
+          !stationReturns.wasHungup && mohStarted && mohStopped &&
+          ds.deviceState === 'INUSE'){
+          done();
+        } else {
+          if (bridgeChannels.length === 2 && !dialed[0].held) {
+            dialed[0].name = 'SIP/phone1';
+            dialed[0].emit('ChannelHold', dialed[0], dialed[0]);
+            stationReturns = getMockChannel();
+            stationReturns.name = 'SIP/phone1';
+            stationReturns.caller = {'number': '5678'};
+            stationReturns.outbound = true;
+            setTimeout(function() {
+              var sla2 = require('../lib/sla.js')(client, config,
+                stationReturns, '42')
+              .catch(errHandler)
+              .done();
+            }, asyncDelay);
+          }
+          extensionUnhold();
+        }
+      }, asyncDelay);
+    }
   });
 });
```

Comment:
For a test, 1095 lines of code is encroaching upon code smell. I think you should investigate ways to break this file up into smaller, more modular components. 

Having too much in one file will cause bloat and make maintenance very challenging.

As a starting place, look for areas where you have almost (if not exactly the same) code. Extract those lines into a function. Remember, one of the first principles is "do not repeat yourself".